### PR TITLE
wis2download test and wis2box downloader commands to manage topics

### PR DESCRIPTION
**Pull request by maaikelimper**
rename 'wis2-downloader' to 'wis2downloader' for consistency in all files and documentation

update test to enable subscribe to self and add test with count check per topic

add 'wis2box downloader' commands to manage subscriptions

improve downloader dashboard and cleanup old dashboards

### CODE REVIEW

**Comment by maaikelimper** [comment]
@david-i-berry this draft PR is used to test your wis2downloader-PR https://github.com/wmo-im/wis2downloader/pull/18

File: wis2box-management/wis2box/downloader.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+@click.pass_context
+@click.option('--topic', '-t', help='The topic to subscribe to', required=True)
+def add_subscription(ctx, topic):
+    """add a subscription"""
+
+    topic = topic.replace('#', '%23')
+    topic = topic.replace('+', '%2B')
+    # make a POST request to http://{DOWNLOAD_URL}/subscriptions
+    try:
+        response = requests.post(f'{DOWNLOAD_URL}/subscriptions?topic={topic}')
```
Comment (david-i-berry) [inline wis2box-management/wis2box/downloader.py:71]: this (?topic={topic}) now needs to be passed in the response body.

File: wis2box-management/wis2box/downloader.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        # check response status
+        if response.status_code == 200:
+            click.echo('Subscription added')
+            click.echo('Current subscriptions:')
+            click.echo(response.text)
```
Comment (david-i-berry) [inline wis2box-management/wis2box/downloader.py:76]: Only the new subscription is returned.

File: wis2box-management/wis2box/downloader.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+@click.pass_context
+@click.option('--topic', '-t', help='The topic to subscribe to', required=True)
+def remove_subscription(ctx, topic):
+    """remove a subscription"""
+
+    topic = topic.replace('#', '%23')
+    topic = topic.replace('+', '%2B')
+    # make a DELETE request to http://{DOWNLOAD_URL}/subscriptions
+    try:
+        response = requests.delete(f'{DOWNLOAD_URL}/subscriptions?topic={topic}')  # noqa
```
Comment (david-i-berry) [inline wis2box-management/wis2box/downloader.py:96]: The topic is now given as part of the url, e.g. 

requests.delete(f'{DOWNLOAD_URL}/subscriptions/cache/a/wis2/%2B/core/%23')

File: wis2box-create-config.py [hunk 4]
```diff
@@ -358,6 +358,17 @@ def create_wis2box_env(config_dir: str) -> None:
...
         fh.write('MINIO_NOTIFY_MQTT_QOS_WIS2BOX=1\n')
         fh.write('\n')
+        # downloader settings
+        fh.write('# downloader settings\n')
+        fh.write('DOWNLOAD_BROKER_HOST=globalbroker.meteo.fr\n')
+        fh.write('DOWNLOAD_BROKER_PORT=8883\n')
+        fh.write('DOWNLOAD_BROKER_USERNAME=everyone\n')
+        fh.write('DOWNLOAD_BROKER_PASSWORD=everyone\n')
+        fh.write('# download transport mechanism (tcp or websockets)\n')
+        fh.write('DOWNLOAD_BROKER_TRANSPORT=tcp\n')
```
Comment (david-i-berry) [inline wis2box-create-config.py:368]: Why tcp and not websockets by default? Port 8883 is often blocked.

File: wis2box-management/wis2box/downloader.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+        click.echo(f'Is the wis2downloader running on {DOWNLOAD_URL}?')
+
+
+@click.command('remove-subscription')
+@click.pass_context
+@click.option('--topic', '-t', help='The topic to subscribe to', required=True)
+def remove_subscription(ctx, topic):
+    """remove a subscription"""
+
+    topic = topic.replace('%23', '#')
```
Comment (david-i-berry) [inline wis2box-management/wis2box/downloader.py:91]: Is this the correct way around?

File: wis2box-management/wis2box/downloader.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    topic = topic.replace('%2B', '+')
```
Comment (david-i-berry) [inline wis2box-management/wis2box/downloader.py:92]: Is this the correct way around?

File: wis2box-create-config.py [hunk 7]
```diff
@@ -358,6 +358,17 @@ def create_wis2box_env(config_dir: str) -> None:
         fh.write('MINIO_NOTIFY_MQTT_TOPIC_WIS2BOX=wis2box/storage\n')
         fh.write('MINIO_NOTIFY_MQTT_QOS_WIS2BOX=1\n')
         fh.write('\n')
+        # downloader settings
```
Comment (david-i-berry) [inline wis2box-create-config.py:361]: What role do the ENV variables have? Should they not be used here?

Comment (maaikelimper) [reply wis2box-create-config.py:361]: I write the default values to the wis2box.env to make it easier for the users to edit them

**Comment by maaikelimper** [comment]
closing, changes have been merged into gts-headers-mapping branch to be reviewed in PR 691

**Pull request closed without merging**